docs/proposals/modifying-output-image.md

```diff
@@ -0,0 +1,147 @@
+
+Allow Build authors to describe modifications they'd like to make to output images.
+For this proposal, we'll scope this down to staticly-defined image [labels](https://docs.docker.com/engine/reference/builder/#label) and [OCI annotations](https://github.com/opencontainers/image-spec/blob/master/annotations.md).
+
+Future proposals might allow Build authors to configure image tags, and/or to allow authors to configure dynamic values based on input values.
+These are non-goals of this proposal.
+
+## Motivation
+
+### Goals
+
+- Begin to enable simple modifications of the output image by Build authors.
+
+### Non-Goals
+
+- Enable dynamic values or placeholders/templating (e.g., git commit SHA)
+- Automatically inject labels or annotations without user input
+- Enable modification of image layer contents (see [runtime image](https://github.com/shipwright-io/build/blob/master/docs/proposals/runtime-image.md))
+
+## Proposal
+
+Add new fields to describe output image labels and annotations:
+
+```yaml=
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: my-build-name
+spec:
+  output:
+    image: quay.io/my/image
+    credentials:
+      name: my-push-creds
+    # New fields
+    labels:
+      maintainer: team@my-company.com
+      description: "This is my cool image"
+    annotations:
+      org.opencontainers.image.url: https://my-company.com/images
+      org.opencontainers.image.source: https://github.com/org/repo
+```
+
+This will be enabled by embedding the existing `Image` struct in a new struct that 
+
+```go=
+type BuildSpec struct {
+    ...
+    Output OutputImage `json:"output"`
+}
+
+type OutputImage struct {
+    Image // embedding existing type
+    
+    // New fields
+    Labels      map[string]string `json:"labels,omitempty"`
+    Annotations map[string]string `json:"annotations,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+When a BuildRun executes for a Build that specifies `labels` or `annotations`, the BuildRun controller will append a new step to the end of the Tekton TaskRun that augments the image with the specified labels and annotations, and pushes to the same tag.
+
+This could be accomplished using a containerized CLI tool like [`crane`](https://github.com/google/go-containerregistry/blob/main/cmd/crane), or a purpose-built binary that Shipwright provides and packages in our releases.
+
+In either case, the TaskRun produced by the BuildRun controller would include a step along the lines of:
+
+```yaml
+- name: modify-output-image
+  image: gcr.io/go-containerregistry/crane  # TODO: pin this to a digest, configurable by Shipwright operators
+  script: |
+    crane mutate $(params.output-image) \
+      --add-label maintaner="team@my-company.com" \
+      --add-label description="This is my cool image" \
+      --add-annotation org.opencontainers.image.url=https://my-company.com/images \
+      --add-annotation org.opencontainers.image.source=https://github.com/org/repo \
```

Comment:
Just a small note: I like Tekton's script support. But, it comes with the price of adding the `place-tools` init container. As a user writing a TaskSpec manually, I would always use `script` if I want to run multiple commands, but we programmatically define the step and could use bash. But, in this specific case, there is imo no need for any of them as just one command runs.

```suggestion
  command:
    - crane
  args:
    - --add-label
    - "maintaner=\"team@my-company.com\""
    - --add-label
    - "description=\"This is my cool image\""
    - --add-annotation
    - "org.opencontainers.image.url=https://my-company.com/images"
    - --add-annotation
    - "org.opencontainers.image.source=https://github.com/org/repo"
```

Reply:
Is there a reason you'd like to avoid the `place-tools` init container? It seems like a fairly small price to pay for the improved readability.

In any case this feels like feedback for the eventual PR 😄

Reply:
`place-scripts`, but I had to edit my comment as well as I mixed it. Well, any init container that can be omitted saves time. Not much I now, but I'm always seeking for the optimum. ;-)